pkg/serverstatusrequest/process.go

```diff
@@ -29,20 +29,23 @@ import (
 	velerov1api "github.com/heptio/velero/pkg/apis/velero/v1"
 	"github.com/heptio/velero/pkg/buildinfo"
 	velerov1client "github.com/heptio/velero/pkg/generated/clientset/versioned/typed/velero/v1"
+	"github.com/heptio/velero/pkg/plugin/clientmgmt"
+	"github.com/heptio/velero/pkg/plugin/framework"
 )
 
 const ttl = time.Minute
 
 // Process fills out new ServerStatusRequest objects and deletes processed ones
 // that have expired.
-func Process(req *velerov1api.ServerStatusRequest, client velerov1client.ServerStatusRequestsGetter, clock clock.Clock, log logrus.FieldLogger) error {
+func Process(req *velerov1api.ServerStatusRequest, client velerov1client.ServerStatusRequestsGetter, pluginRegistry clientmgmt.Registry, clock clock.Clock, log logrus.FieldLogger) error {
```

Comment:
For testing purposes, I would probably do the following:

1. Define a new interface in this package, along the lines of:
```go
type PluginLister interface {
  List(...) // TODO fill in parameters and return types to match plugin.Registry
}
```

2. Switch the `pluginRegistry` parameter to this function to be `pluginLister PluginLister`

3. For testing, implement a simple `fakePluginLister` struct that implements the `PluginLister` interface and returns whatever you tell it to, and pass that as the argument to `Process(...)`.

I'm still thinking about some ideas for making it easier to unit test code that relies on plugins, but for now I think this is a reasonable approach.

Reply:
Great, thank you!